src/netconf_server_ssh.c

```diff
@@ -92,19 +92,48 @@ np2srv_pubkey_auth_cb(const struct nc_session *session, ssh_key key, void *UNUSE
     const char *username;
     char *line = NULL, *ptr, *ptr2;
     size_t n;
-    int r, ret = 1, line_num = 0;
+    int r, ret = 1, line_num = 0, invalid_username = 0;
 
     username = nc_session_get_username(session);
 
     errno = 0;
     pwd = getpwnam(username);
-    if (!pwd) {
+    if (!NP2SRV_SSH_AUTHORIZED_KEYS_ARG_IS_USERNAME && !pwd) {
         ERR("Failed to find user entry for \"%s\" (%s).", username, errno ? strerror(errno) : "User not found");
         goto cleanup;
     }
 
+    /* if the user is not a system user, we need at least a bit of validation */
```

Comment:
Please put the validation into a separate function.

Reply:
done.